src/runtime_src/core/tools/xbmgmt2/SubCmdConfigure.cpp

```diff
@@ -0,0 +1,503 @@
+    loadConfigOptions.add_options()
+      ("input", boost::program_options::value<decltype(path)>(&path),"INI file with the memory configuration")
+    ;
+
+    // Options previously under the config command
+    po::options_description configOptions("Config Options");
+    configOptions.add_options()
+      ("retention", boost::program_options::value<decltype(retention)>(&retention),"Enables / Disables memory retention.  Valid values are: [ENABLE | DISABLE]")
+    ;
+
+    po::options_description commonOptions("Common Options");
+    commonOptions.add_options()
+        ("device,d", boost::program_options::value<decltype(devices)>(&devices)->multitoken(), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest")
+        ("help,h", boost::program_options::bool_switch(&help), "Help to use this sub-command")
+    ;
+
+    commonOptions.add(loadConfigOptions);
+    commonOptions.add(configOptions);
+
+    // Hidden options previously under the config command
+    po::options_description configHiddenOptions("Hidden Options");
+    configHiddenOptions.add_options()
+        ("daemon", boost::program_options::bool_switch(&daemon), "Update the device daemon configuration")
+        ("host", boost::program_options::value<decltype(host)>(&host), "IP or hostname for device peer")
+        ("security", boost::program_options::value<decltype(security)>(&security), "Update the security level for the device")
+        ("runtime_clk_scale", boost::program_options::value<decltype(clk_scale)>(&clk_scale), "Enable/disable the device runtime clock scaling")
+        ("cs_threshold_power_override", boost::program_options::value<decltype(power_override)>(&power_override), "Update the power threshold in watts")
+        ("cs_threshold_temp_override", boost::program_options::value<decltype(temp_override)>(&temp_override), "Update the temperature threshold in celsius")
+        ("cs_reset", boost::program_options::value<decltype(cs_reset)>(&cs_reset), "Reset all scaling options")
+        ("showx", boost::program_options::bool_switch(&showx), "Display the device configuration settings")
+    ;
+
+    po::options_description hiddenOptions("Hidden Options");
+    hiddenOptions.add(configHiddenOptions);
+
+    po::options_description allOptions("All Options");
+    allOptions.add(commonOptions);
+    allOptions.add(hiddenOptions);
+
+    // Parse sub-command ...
+    po::variables_map vm;
+
+    try {
+        po::store(po::command_line_parser(_options).options(allOptions).run(), vm);
+        po::notify(vm); // Can throw
+    } catch (po::error& e) {
+        std::cerr << "ERROR: " << e.what() << std::endl << std::endl;
+        printHelp(commonOptions, hiddenOptions);
+        throw xrt_core::error(std::errc::operation_canceled);
+    }
+
+    // Ensure mutual exclusion amongst the load config and config options
+    // TODO Once all of the config options are incorporated into the load config file
```

Comment:
Is this TODO comment still needed?

Reply:
The hidden options will eventually be brought into the config file parsing. At the moment they are not parsed so I added the comment